api/src/test/java/org/openmrs/api/OrderServiceTest.java

```diff
@@ -3794,12 +3800,139 @@ public void saveOrderGroup_shouldFailValidationIfAnyOrdersFailValidation() {
 	}
+	
+	@Test
+	public void getOrderGroupAttributeTypes_shouldReturnAllOrderGroupAttributeTypes(){
+		executeDataSet(ORDER_GROUP_ATTRIBUTES);
+		List<OrderGroupAttributeType>orderGroupAttributeTypes=orderService.getOrderGroupAttributeTypes();
+		assertEquals(4,orderGroupAttributeTypes.size());
+	}
+	
+	@Test
+	public void getOrderGroupAttributeType_shouldReturnNullIfNonExistingIdIsProvided(){
+		executeDataSet(ORDER_GROUP_ATTRIBUTES);
+		assertNull(Context.getOrderService().getOrderGroupAttributeTypeById(10));
+	}
+
+	@Test 
+	public void getOrderGroupAttributeType_shouldReturnOrderGroupAttributeTypeGivenId(){
+		executeDataSet(ORDER_GROUP_ATTRIBUTES);
+		final Integer ID = 2;
+		final String UUID2="9cf1bbe6-d18e-11ea-87d0-0242ac130003";
+		OrderGroupAttributeType orderGroupAttributeType = orderService.getOrderGroupAttributeTypeById(ID);
+		assertEquals(orderService.getOrderGroupAttributeTypeByUuid(UUID2),orderGroupAttributeType);
+		assertEquals(orderGroupAttributeType.getUuid(),Context.getOrderService().getOrderGroupAttributeTypeByUuid(UUID2).getUuid());
+		assertEquals(ID,Context.getOrderService().getOrderGroupAttributeTypeByUuid(UUID2).getId());
+	}
+
+	@Test
+	public void getOrderGroupAttributeTypeByUuid_shouldReturnOrderGroupAttributeTypeByUuid(){
+		executeDataSet(ORDER_GROUP_ATTRIBUTES);
+		OrderGroupAttributeType orderGroupAttributeType = orderService.getOrderGroupAttributeTypeByUuid("9cf1bce0-d18e-11ea-87d0-0242ac130003");
+		assertEquals("Bacteriology",orderGroupAttributeType.getName());
+	}
+
+	@Test
+	public void saveOrderGroupAttributeType_shouldSaveOrderGroupAttributeTypeGivenOrderGroupAttributeType() throws ParseException {
+		executeDataSet(ORDER_GROUP_ATTRIBUTES);
+		int initialGroupOrderAttributeTypeCount = Context.getOrderService().getOrderGroupAttributeTypes().size();
+		OrderGroupAttributeType orderGroupAttributeType = new OrderGroupAttributeType();
+		orderGroupAttributeType.setName("Surgery");
+		orderGroupAttributeType.setDatatypeClassname(FreeTextDatatype.class.getName());
+		Context.getOrderService().saveOrderGroupAttributeType(orderGroupAttributeType);
+		assertNotNull(orderGroupAttributeType.getId());
+		assertEquals(initialGroupOrderAttributeTypeCount+1,Context.getOrderService().getOrderGroupAttributeTypes().size());
+	}
+		
+	@Test
+	public void saveOrderGroupAttributeType_shouldEditAnExistingOrderGroupAttributeType(){
+		executeDataSet(ORDER_GROUP_ATTRIBUTES);
+		final String UUID4 ="9cf1bdb2-d18e-11ea-87d0-0242ac130003";
+		final String name ="ECG";
+		//Check for values in the database
+		OrderGroupAttributeType orderGroupAttributeType=Context.getOrderService().getOrderGroupAttributeTypeById(4);
+		assertEquals(UUID4,orderGroupAttributeType.getUuid());
+		assertEquals(name,orderGroupAttributeType.getName());
+		assertEquals("Testing unretire",orderGroupAttributeType.getRetireReason());
+        //edit existing values in the database
```

Comment:
The indention seems off.

Reply:
This was fixed

Reply:
I fixed the indent and been routinely running "mvn clean install "  before each commit , am however not sure why the local indentg is not reflected here . This is how my local indenting looks like.
![DeepinScreenshot_select-area_20200908052632](https://user-images.githubusercontent.com/16301386/92427115-4bf4a700-f194-11ea-9053-fa2744ac1c8c.png)
![DeepinScreenshot_select-area_20200908052533](https://user-images.githubusercontent.com/16301386/92427123-52831e80-f194-11ea-9176-b1460b4bb2f5.png)
![DeepinScreenshot_select-area_20200908052600](https://user-images.githubusercontent.com/16301386/92427133-5878ff80-f194-11ea-8f86-6081cc835d95.png)

Reply:
But am working to see that the changes are reflected.